addons/mail/tests/common.py

```diff
@@ -1462,9 +1462,15 @@ def _activate_multi_lang(cls, lang_code='es_ES', layout_arch_db=None, test_recor
             cls.env['ir.model']._get(test_record._name).with_context(lang=lang_code).name = 'Spanish Model Description'
 
         # Translate some code strings used in mailing
-        code_translations.python_translations[('test_mail', 'es_ES')]['NotificationButtonTitle'] = 'SpanishButtonTitle'
+        code_translations.python_translations[('test_mail', 'es_ES')] = {
+            **code_translations.python_translations[('test_mail', 'es_ES')],
+            'NotificationButtonTitle': 'SpanishButtonTitle'
+        }
         cls.addClassCleanup(code_translations.python_translations[('test_mail', 'es_ES')].pop, 'NotificationButtonTitle')
-        code_translations.python_translations[('mail', 'es_ES')]['View %s'] = 'SpanishView %s'
+        code_translations.python_translations[('mail', 'es_ES')] = {
+            **code_translations.python_translations[('mail', 'es_ES')],
+            'View %s': 'SpanishView %s'
+        }
```

Comment:
Doesn't this then transform python_translation into a non-readonly dict, leading to the same issue?

Maybe the clean-up should set `python_translations` directly, again.
Or maybe we can patch some method for the duration of the test instead of modifying a global here, may be safer